src/PerfView/PerfViewData.cs

```diff
@@ -4122,36 +4123,54 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
             }
             else if (streamName == "CCW Ref Count")
             {
-                // TODO use the callback model.  We seem to have an issue getting the names however. 
-                foreach (var data in events.ByEventType<CCWRefCountChangeTraceData>())
+                var interopTraceEventParser = new InteropTraceEventParser(eventSource);
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWQueryRuntimeClassNameArgs>(args =>
                 {
                     sample.Metric = 1;
-                    sample.TimeRelativeMSec = data.TimeStampRelativeMSec;
-                    var stackIndex = stackSource.GetCallStack(data.CallStackIndex(), data);
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
 
-                    var operation = data.Operation;
-                    if (operation.StartsWith("Release", StringComparison.OrdinalIgnoreCase))
-                        sample.Metric = -1;
+                    var ccwRefKindName = "CCW QueryRuntimeClassName";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWRefCountIncArgs>(args => {
+                    sample.Metric = 1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
 
-                    var ccwRefKindName = "CCW " + operation;
-                    var ccwRefKindIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
-                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefKindIndex, stackIndex);
+                    var ccwRefKindName = "CCW RefCountInc";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
```

Comment:
These frames don't actually show up on the view, because you did not do stackIndex = ....Intern(... stackIndex).  
This error was made several places.

Reply:
done